### PR TITLE
Feature/473 position margin levels

### LINKED ISSUE #473: Position margin level
As a user of console I want to see how close I am to being liquidated.

#### Acceptance criteria
- [x] Show estimated liquidation price
- [x] Show entry price
- [x] Show progress bar indicating how close you are to the liquidation price
- [x] Color should change using Intent to indicate 'danger' level using @witgaw calculations below

### PR DESCRIPTION
**Pull request by bglownia**
# Related issues 🔗

Closes #473

### CODE REVIEW

File: libs/positions/src/lib/positions-metrics-data-provider.ts [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    const marginMaintenance = new BigNumber(
+      marginLevel.maintenanceLevel
+    ).multipliedBy(marketDecimalPlaces);
+    const marginSearch = new BigNumber(marginLevel.searchLevel).multipliedBy(
+      marketDecimalPlaces
+    );
+    const marginInitial = new BigNumber(marginLevel.initialLevel).multipliedBy(
+      marketDecimalPlaces
+    );
+    /*
```
Comment (mattrussell36) [inline libs/positions/src/lib/positions-metrics-data-provider.ts:193]: chore (blocking): Remove this?

File: libs/positions/src/lib/positions-metrics-data-provider.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+    metrics.push({
+      marketName: market.name,
+      averageEntryPrice: position.node.averageEntryPrice,
+      capitalUtilisation: Math.round(capitalUtilisation.toNumber()),
+      currentLeverage: currentLeverage.toNumber(),
+      marketDecimalPlaces,
+      positionDecimalPlaces,
+      assetDecimals,
+      // generalAccountBalance: generalAccount?.balance ?? '0',
```
Comment (mattrussell36) [inline libs/positions/src/lib/positions-metrics-data-provider.ts:226]: chore (blocking): can we tidy up these comments?

File: libs/positions/src/lib/positions-table.spec.tsx [hunk 3]
```diff
@@ -1,103 +1,196 @@
...
-        expect(headers.map((h) => h.textContent?.trim())).toEqual([
-          'Market',
-          'Amount',
-          'Average Entry Price',
-          'Mark Price',
-          'Realised PNL',
-        ]);
-      });
+it('Splits market name', async () => {
+  act(async () => {
```
Comment (mattrussell36) [inline libs/positions/src/lib/positions-table.spec.tsx:73]: question (blocking): Not sure `act` usage is correct in these tests? You should only have to wrap the code that causes the update, in this case render (due to ag grid)

```
await act(async () => {
  render(...)
})
expect(screen.getByText('ETH/BTC')).toBeTruthy();
expect(screen.getByText('31 july 2022')).toBeTruthy();
```

Comment (mattrussell36) [reply libs/positions/src/lib/positions-table.spec.tsx:73]: _I think_ that this should also work and is a bit cleaner with less pyramid of doom.
```
render(...)
expect(await screen.findByTest('ETH/BTC')).toBeTruthy()
```

File: libs/positions/src/lib/positions-table.tsx [hunk 4]
```diff
@@ -1,168 +1,328 @@
...
+        cellRenderer={MarketNameCell}
+        valueFormatter={({
+          value,
+        }: PositionsTableValueFormatterParams & {
+          value: Position['marketName'];
+        }) => {
+          if (!value) {
+            return undefined;
+          }
+          const matches = value.match(/^(.*)\((.*)\)\s*$/);
```
Comment (mattrussell36) [inline libs/positions/src/lib/positions-table.tsx:143]: chore (non blocking): Whats going on her? Could you add a comment

File: libs/positions/tailwind.config.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const { join } = require('path');
```
Comment (mattrussell36) [inline libs/positions/tailwind.config.js:1]: question (non blocking): I didn't think libs needed their owntailwind config? Is this required to get styles for storybook working?

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/positions/tailwind.config.js (hunk 5 above)
Comment (bglownia) [reply libs/positions/tailwind.config.js:1]: Tailwind config is only for storybook.

**Review by mattrussell36 — APPROVED** [review]

**Review by asiaznik — APPROVED** [review]

File: libs/positions/src/lib/positions-metrics-data-provider.ts [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    );
+    const assetDecimals = marginAccount.asset.decimals;
+    const { positionDecimalPlaces, decimalPlaces: marketDecimalPlaces } =
+      market;
+    const openVolume = new BigNumber(position.node.openVolume).dividedBy(
+      10 ** positionDecimalPlaces
+    );
+
+    const marginAccountBalance = marginAccount
+      ? new BigNumber(marginAccount.balance).dividedBy(10 ** assetDecimals)
```
[the remark this answers is not in the document]
Comment (bglownia) [reply libs/positions/src/lib/positions-metrics-data-provider.ts:154]: I will improve this in https://github.com/vegaprotocol/frontend-monorepo/pull/1047

**Pull request merged**